compiler/optimizer/optimizer.go

```diff
@@ -185,6 +185,10 @@ func (o *Optimizer) OptimizeDeleter(seq dag.Seq, replicas int) (dag.Seq, error)
 	return dag.Seq{lister, scatter, merge}, nil
 }
 
+func (o *Optimizer) SortKeys(seq dag.Seq) ([]order.SortKey, error) {
+	return o.propagateSortKey(copyOps(seq), []order.SortKey{order.Nil})
+}
+
```

Comment:
Nit: Put this right before propagateSortKey since it's just a wrapper around that?